libr/core/project.c

```diff
@@ -774,6 +772,14 @@ R_API bool r_core_project_save(RCore *core, const char *prj_name) {
 	char *old_prj_name = NULL;
 	r_return_val_if_fail(prj_name && *prj_name, false);
 	script_path = get_project_script_path (core, prj_name);
+	if (core->bin->is_debugger) {
+		eprintf ("radare2 does not support projects on debugged bins\n");
+		return false;
+	}
+	if (r_config_get_i (core->config, "asm.emu")) {
```

Comment:
this variable just makes the disasm emulate the selected instructions to show computed references. i dont see why an emulated binary shouldnt be a valid project. Also, what is the problem for not supporting debug/emulation sessions? both are very common use cases and saving/restoring files and flags and anal info shouldnt be affected by it.

In any case i'm fine for forbiddnig to save a project when in debugger mode. but i dont see why the emulated environment shouldnt be supported. (the only reason is missing iomaps for stack and other stuff like that) and this is the main reason why we dont support projects well because rbin doesnt hold this information and core cant corrlate io and sort its operational orders. and refactoring io and bin are max prio in order to support any kind of environment configured in r2 as in projects.

Reply:
I see your point and I will remove the lines that forbid emulation but I would like to say that when it comes down to the general case emulation and projects are broken together (right now at least). We can allow emulation when a work around has been found.

Reply:
It is worth pointing out that even though saving a debug project is in fact very useful, it is currently broken so there would be no point in giving the user a false sense of comfort that their "project" is saved and can be reopend easily.